File: problems/SRM-144-DIV-2/200/Time.cc

// {{{ VimCoder 0.3.5 <-----------------------------------------------------
// vim:filetype=cpp:foldmethod=marker:foldmarker={{{,}}}

#include <algorithm>
#include <bitset>
#include <cmath>
#include <cstdio>
#include <cstdlib>
#include <ctime>
#include <deque>
#include <functional>
#include <iomanip>
#include <iostream>
#include <list>
#include <map>
#include <numeric>
#include <queue>
#include <set>
#include <sstream>
#include <stack>
#include <utility>
#include <vector>

using namespace std;

// }}}

class Time {
public:
  string whatTime(int seconds) {
    string res;

    for (int base = 3600; base > 0; base /= 60) {
      res += to_string(seconds / base);

      if (base > 1)
        res += ":";

      seconds %= base;
    }
    
    return res;
  }
};

File: problems/SRM-144-DIV-2/200/Time.java

// {{{ VimCoder 0.3.5 <-----------------------------------------------------
// vim:filetype=java:foldmethod=marker:foldmarker={{{,}}}

import static java.lang.Math.*;
import static java.math.BigInteger.*;
import static java.util.Arrays.*;
import static java.util.Collections.*;
import java.math.*;
import java.util.*;

// }}}

public class Time {
	public String whatTime(int seconds) {
        String res = "";

        for (int base = 3600; base > 0; base /= 60) {
            res += seconds / base;
            if (base > 1)
                res += ":";

            seconds %= base;
        }

        return res;
	}
}
